File: tests/go/cases/case_087_reconcile_session.go
package escape_tests

var retainedCase087 = []map[string]string{}

func Case087ReconcileSession(input string) string {
    // Task case 087: reconcile session payload for downstream business processing.
    raw := input
    if raw == "" {
        raw = "sample"
    }
    payload := map[string]string{
        "task": "reconcile_session",
        "entity": "session",
        "stage": "reconcile",
        "input": raw,
    }
    // ESCAPE: payload is copied into retained audit sink.
    retainedCase087 = append(retainedCase087, payload)
    return "ok"
}

File: tests/rust/cases/case_087_reconcile_session.rs
#![allow(unused)]

use std::collections::HashMap;
use std::sync::{Mutex, OnceLock};

#[allow(dead_code)]
static RETAINED_CACHE: OnceLock<Mutex<Vec<HashMap<String, String>>>> = OnceLock::new();
#[allow(dead_code)]
static RETAINED_AUDIT: OnceLock<Mutex<Vec<HashMap<String, String>>>> = OnceLock::new();
#[allow(dead_code)]
static RETAINED_HANDLERS: OnceLock<Mutex<Vec<String>>> = OnceLock::new();

pub fn case_087_reconcile_session(input: String) -> String {
    // Task case 087: reconcile session payload for downstream business processing.
    let task_name = "reconcile_session".to_string();
    let raw = if input.is_empty() { "sample".to_string() } else { input };
    let mut payload: HashMap<String, String> = HashMap::new();
    payload.insert("task".to_string(), task_name.clone());
    payload.insert("entity".to_string(), "session".to_string());
    payload.insert("stage".to_string(), "reconcile".to_string());
    payload.insert("input".to_string(), raw.clone());
    // ESCAPE: payload is copied into retained audit sink.
    RETAINED_AUDIT.get_or_init(|| Mutex::new(Vec::new())).lock().expect("audit lock").push(payload);
    "ok".to_string()
}
